### ISSUE TITLE
To get cam_intrinsics and cam_extrinsics from .npz files

### ISSUE THREAD

**Issue by MoyGcc** [Arthur151/ROMP#300]
Hi Yu, thanks for your work and such an organized repo! 

I'm now using ROMP to get SMPL poses and would like to visualize the meshes via a perspective camera. I usually use a similar way as https://github.com/chungyiweng/humannerf/issues/1 to convert the `s`, `t_x`, and `t_y` along with a human bbox to the pinhole camera parameters and it does work on VIBE output. However, it seems like I cannot easily get the parameters with the output from ROMP .npz outputs (I can get a rough bbox from `pj2d_org` ). I found that the scaling factor `s` is quite different between the VIBE and ROMP estimation for the same input image (~1.14 in VIBE and ~0.58 in ROMP). Could you please point out how I can quickly obtain (estimate) the camera intrinsic and extrinsic? Thanks!

**Comment by hongsiyu** [comment]
> Hi Yu, thanks for your work and such an organized repo!
> 
> I'm now using ROMP to get SMPL poses and would like to visualize the meshes via a perspective camera. I usually use a similar way as [chungyiweng/humannerf#1](https://github.com/chungyiweng/humannerf/issues/1) to convert the `s`, `t_x`, and `t_y` along with a human bbox to the pinhole camera parameters and it does work on VIBE output. However, it seems like I cannot easily get the parameters with the output from ROMP .npz outputs (I can get a rough bbox from `pj2d_org` ). I found that the scaling factor `s` is quite different between the VIBE and ROMP estimation for the same input image (~1.14 in VIBE and ~0.58 in ROMP). Could you please point out how I can quickly obtain (estimate) the camera intrinsic and extrinsic? Thanks!

Have you solved this problem? I met the same issue.

**Comment by MoyGcc** [comment]
Hi Yu @Arthur151,
Thanks so much for the quick reply and for pointing out the correct way to do this. In the end, I followed the way that you applied for evaluation on AGORA: https://github.com/Arthur151/ROMP/blob/91dac0172c4dc0685b97f96eda9a3a53c626da47/simple_romp/evaluation/eval_AGORA.py#L79 
and now the projected smpl mesh can align well with the image. Though there is still a "slight" difference (below, the one with normal color is my projected result) in terms of the projection. I think it's okay. @hongsiyu, you could probably also refer to the evaluation on the AGORA dataset for doing this.

![0000](https://user-images.githubusercontent.com/29098423/177960677-b32c6455-6019-4345-8ad6-6a41c82c059f.png)
![00000000](https://user-images.githubusercontent.com/29098423/177960695-2f7f6f51-128b-462e-ae1a-df18c13967d9.png)

**Comment by Arthur151** [comment]
That's clever. Glad to hear that.

**Comment by Andyen512** [comment]
> 

So the intrinsic is ([443.4, 1, 512//2], [1, 443.4, 512//2],[0, 0, 1]), and the extrinsics[:3, 3] = cam_trans , right?

**Comment by Andyen512** [comment]
So the intrinsic is ([443.4, 1, 512//2], [1, 443.4, 512//2],[0, 0, 1]), and the extrinsics[:3, 3] = cam_trans , right? @MoyGcc

**Comment by hongsiyu** [comment]
> Hi Yu @Arthur151, Thanks so much for the quick reply and for pointing out the correct way to do this. In the end, I followed the way that you applied for evaluation on AGORA:
> 
> https://github.com/Arthur151/ROMP/blob/91dac0172c4dc0685b97f96eda9a3a53c626da47/simple_romp/evaluation/eval_AGORA.py#L79
> 
> 
> and now the projected smpl mesh can align well with the image. Though there is still a "slight" difference (below, the one with normal color is my projected result) in terms of the projection. I think it's okay. @hongsiyu, you could probably also refer to the evaluation on the AGORA dataset for doing this.
> ![0000](https://user-images.githubusercontent.com/29098423/177960677-b32c6455-6019-4345-8ad6-6a41c82c059f.png) ![00000000](https://user-images.githubusercontent.com/29098423/177960695-2f7f6f51-128b-462e-ae1a-df18c13967d9.png)

I followed the way you mentioned with my own video. But the progress image in humannerf  seems not correct. Do you succeed in trainiing humannerf with AGORA dataset.

**Comment by Arthur151** [comment]
@Andyen512 
No, The image size should be the original size on input image, not on the resize BEV's input map

**Comment by hongsiyu** [comment]
> @Andyen512 No, The image size should be the original size on input image, not on the resize BEV's input map. It is fine to directly use the camera intrinsic in humannerf during calculating the 3D translation using `estimate_translation`
> 
> ```
> "cam_intrinsics": [
>             [23043.9, 0.0,940.19],
>             [0.0, 23043.9, 539.23],
>             [0.0, 0.0, 1.0]
> ```

Thank you very much, the focal length makes me succeed in training humannerf.

**Comment by Arthur151** [comment]
@Andyen512 
That focal length (23043.9) / image center coords (940.19, 539.23) is just for training humannerf in their camera extrinsic matrix.

To inference on you own video, you can re-calculate the focal length :
when FOV=60 deg, focal length = H/2 * 1/(tan(FOV/2)) = 1920/2. * 1./np.tan(np.radians(30)) = 1662.768

**Comment by Andyen512** [comment]
ok thx, I'll try

**Comment by mch0dmin** [comment]
> length

hi @hongsiyu , can you tell me how to use ROMP to obtain "3x3" cam_intrinsics and "4x4" cam_extrinsics, thanks.
